doc/SDK_cross_compile_example.md

```diff
@@ -61,50 +61,61 @@ In the above command replace &lt;*your Pi identifier*&gt; with the IP address of
 ```
-Using the text editor of your choice, create a new file in this directory and call it toolchain-rpi.cmake. Into this file place the following lines
-
-```cmake
-INCLUDE(CMakeForceCompiler)
-
-SET(CMAKE_SYSTEM_NAME Linux)     # this one is important
-SET(CMAKE_SYSTEM_VERSION 1)     # this one not so much
-
-# this is the location of the amd64 toolchain targeting the Raspberry Pi
-SET(CMAKE_C_COMPILER $ENV{RPI_ROOT}/../bin/arm-linux-gnueabihf-gcc)
-
-# this is the file system root of the target
-SET(CMAKE_FIND_ROOT_PATH $ENV{RPI_ROOT})
-
-# search for programs in the build host directories
-SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-
-# for libraries and headers in the target directories
-SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+Now to create the toolchain and build the SDK, run the script found at:
 ```
-and save the toolchain file. Your cross compilation environment is now complete.
-
-### Building the SDK
-
-The final step in the process is to run the actual build. For this you will need to be in the Linux build directory as shown above. Enter the following commands
+/Source/azure-iot-sdk-c/blob/main/jenkins/raspberrypi_c_buster.sh
 ```
-cd ~/Source/azure-iot-sdk-c/build_all/linux
-./build.sh --toolchain-file toolchain-rpi.cmake -cl --sysroot=$RPI_ROOT
+
 ```
-This will tell cmake to build the SDK using the toolchain file toolchain-rpi.cmake. Finally, and absolutely critical is the use of the *--sysroot* option. Without this the compiler will fail to find required headers and libraries.
+#!/bin/bash
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+set -x # Set trace on
+set -o errexit # Exit if command failed
+set -o nounset # Exit if variable not set
+set -o pipefail # Exit if pipe failed
+
+cd ~/Source/azure-iot-sdk-c/
+
+# assume directory is root of downloaded repo
+mkdir cmake
+cd cmake
+ls -al
+
+# Create a cmake toolchain file on the fly
+echo "SET(CMAKE_SYSTEM_NAME Linux)     # this one is important" > toolchain.cmake
+echo "SET(CMAKE_SYSTEM_VERSION 1)      # this one not so much" >> toolchain.cmake
+
+echo "SET(CMAKE_C_COMPILER ${TOOLCHAIN_EXES}/${TOOLCHAIN_NAME}-gcc)" >> toolchain.cmake
+echo "SET(CMAKE_CXX_COMPILER ${TOOLCHAIN_EXES}/${TOOLCHAIN_NAME}-g++)" >> toolchain.cmake
+echo "SET(CMAKE_FIND_ROOT_PATH ${TOOLCHAIN_SYSROOT})" >> toolchain.cmake
+echo "SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> toolchain.cmake
+echo "SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> toolchain.cmake
+echo "SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)" >> toolchain.cmake
+ls -al
+
+# Build the SDK. This will use the OpenSSL, cURL and uuid binaries that we built before
+cmake -DCMAKE_TOOLCHAIN_FILE=toolchain.cmake -Duse_prov_client:BOOL=OFF -DCMAKE_INSTALL_PREFIX=${TOOLCHAIN_PREFIX} -Drun_e2e_tests:BOOL=ON -Drun_unittests=:BOOL=ON ..
+make -j 2
+ls -al
```

Comment:
I would remove all of this since it could become outdated from the script that you linked. Might as well just have one source of truth

Reply:
Also, this is trying to run e2e tests AND unit tests. Also this does not compile the prov_client. This needs to build the prov_client like the other cross-compile samples.

Reply:
@danewalton So just put a link to the script?

Reply:
Yes the script has the same thing which you posted so it's best to have one place for it